### PR TITLE
56 capture a piece

### PR DESCRIPTION
**Pull request by JamesH34**
Updated Put view and Get view in views.py in the games_app. Players are able to capture pieces, and the related Cypress test passes.

### CODE REVIEW

File: frontend/cheapchess/src/components/Board.js [hunk 1]
```diff
@@ -33,23 +33,25 @@ const Board = (
...
-    } else {
+    // else
+    //  if (
+    //   parentState.gameDataFromServer?.whose_turn &&
+    //   parentState.gameDataFromServer.whose_turn.id !== parentState.auth.user.id
+    //   )
+    // {
+    //   parentState.setMessages({'Nice try' : 'wait your turn!'});
+    // /* It IS our turn.  Perform additional validation */
+    // } else {
```
Comment (chsbca) [inline frontend/cheapchess/src/components/Board.js:54]: did we want lines 38 to 54 commented out? This is the check that @macleanb put to verify if it's the player's turn to move or check if there is a second player to start the match with

**Review by chsbca — CHANGES_REQUESTED** [review]
image

trying to do some testing to find bugs, and i found one.
To give context for the image above, on white's turn, they had a white rook on the upper left corner. I proceeded to capture black's rook right below it, but the game/database reflects the black rook on that spot but the white rook being gone, despite being the attacker.
It is currently black rook's turn, and upon piece selection, the database does not return any legal moves it can make, despite it having paths to do so. I think the confusion of the database and the frontend might be the reason why there's this weird bug.
I was able to consistently replicate the bug (3 times to be specific), and it seems to be the rank A rooks capturing each other having some weird behavior there

**Comment by chsbca** [comment]
Additional info on the bug I found, it seems to be some weird behavior with white capturing black. Once a move is made from white going to capture black's piece, the bug seems to be that sometimes black will come out on top.

**Comment by macleanb** [comment]
I wrote a test (1c) on feature branch #67 if you want to check it out.  Branch 67 merged (includes) branch 56, but I'm not seeing this test reproducing the buggy behavior.

File: frontend/cheapchess/src/components/Board.js (hunk 1 above)
Comment (macleanb) [reply frontend/cheapchess/src/components/Board.js:54]: Good catch Chris.  We do not want those lines commented out.  We need to uncomment them and re-commit.  Thanks!

**Comment by chsbca** [comment]
I'll try to replicate the behavior on my end too come tomorrow morning. I'll use a screen recording software too to hopefully capture that behavior and see if yall can replicate it and check if it produces the same results.

**Comment by macleanb** [comment]
I have a hunch.  In Issue #67 (the test file) I added some print statements to illustrate what I suspect is happening.

When we do:
```            
captured_piece = updated_game.pieces.filter(
                current_file = destination_square_id[0],
                current_rank = destination_square_id[1]
            ).exclude(pk=updated_piece.id).first()
```
...I suspect that sometimes more than one piece is somehow remaining in the filter result, despite our attempts to exclude the piece we just selected.  I'm no expert on exclude, but I wonder if it isn't working the way we expected.

So I wrote a test below that:
```
            # test
            test_captured_pieces = updated_game.pieces.filter(
                current_file = destination_square_id[0],
                current_rank = destination_square_id[1]
            ).exclude(pk=updated_piece.id)
            print('Here in games_app views.py, number of captured pieces: ')
            print(test_captured_pieces.count())
```

Sure enough -- on a few occasions when pieces are captured, I see "2" printed out once in a while (with knights and pawns, but not rooks), indicating that the filtered captured pieces have more than one object even though we tried to filter out our piece that moved.  This doesn't always happen.  But when it does, calling .first() on the filtered pieces will probably return a light or a dark piece randomly.

We could read up on the exclude method, or instead just include the opponent's color in the filter and get rid of exclude() altogether.

**Comment by JamesH34** [comment]
Feature is complete in accordance with specification. Passing Cypress tests.
Screenshot 2024-05-01 at 12 15 15 PM

**Review by chsbca — APPROVED** [review]

**Pull request merged**